interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java

```diff
@@ -300,8 +302,44 @@ public void sendHeaders(Metadata responseHeaders) {
       };
       ServerCall.Listener<ReqT> noopListener = new ServerCall.Listener<ReqT>() {};
 
-      // sleep if instructed by rpc-behavior
-      for (String callBehavior : callBehaviors) {
+      int attemptNum = 0;
+      String attemptNumHeader = requestHeaders.get(ATTEMPT_NUM);
+      if (attemptNumHeader != null) {
+        try {
+          attemptNum = Integer.valueOf(attemptNumHeader);
+        } catch (NumberFormatException e) {
+          newCall.close(
+              Status.INVALID_ARGUMENT.withDescription(
+                  String.format(
+                      "Invalid format for grpc-previous-rpc-attempts header (%s)",
+                      attemptNumHeader)),
+              new Metadata());
+          return noopListener;
+        }
+      }
+
+      for (String callBehaviorEntry : callBehaviors) {
+        String callBehavior;
```

Comment:
I suggest we don't have this, and just reassign `callBehaviorEntry` when appropriate (and probably rename it to `callBehavior`).

Reply:
Done. That was a habit from using `const` variables by default in JS.